modules/agrvate/main.nf

```diff
@@ -0,0 +1,38 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process AGRVATE {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::agrvate=1.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/agrvate:1.0--hdfd78af_0"
+    } else {
+        container "quay.io/biocontainers/agrvate:1.0--hdfd78af_0"
+    }
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path("${fasta.baseName}-results"), emit: results_dir
```

Comment:
I think the summary.tab should also be emited since that's the most likely file used downstream (e.g. merging results from multiple samples)

```
tuple val(meta), path("${fasta.baseName}-results/${fasta.baseName}-summary.tab"), emit: summary
```

Reply:
Makes sense, I have accomodated this now.